src/datasets/arrow_dataset.py

```diff
@@ -381,6 +381,16 @@ def to_tf_dataset(
         else:
             raise ImportError("Called a Tensorflow-specific function but Tensorflow is not installed.")
 
+        warnings.warn(
```

Comment:
Cool thanks !

Maybe we can warn only if the user passes a list with one element ?

Reply:
Certainly :) I originally had it like that, and wasn't sure if it was better to let everyone know. Your vote tips the balance - I'll update 👍